packages/hdwallet-core/src/ethereum.ts

```diff
@@ -58,6 +58,8 @@ export interface ETHSignTx {
    * Device must `ethSupportsNativeShapeShift()`
    */
   exchangeType?: ExchangeType;
+  type: string,
+  accessList?:[]
```

Comment:
Using a `string` here isn't a great choice, and from the field name I can't even tell what you mean! If you're talking about the EIP-2718 transaction type, that's at a lower level of abstraction than is appropriate here. The correct TX type is inferred from whether `gasPrice` or `maxFeePerGas`/`maxPriorityFeePerGas` is present; including an explicit flag for it would make the type denormalized.

Happy to support `accessList` here -- though that's really EIP-2930, not EIP-1559 -- but that type `[]` definitely isn't correct. Maybe `[string, string[]][]` with a doc comment indicating that those refer to addresses / storage keys respectively? In any case, we'll need to see an implementation in at least `hdwallet-native` as well as accompanying unit and integration tests, to move forward.

Reply:
I appreciate the feedback and I will make some improvements.